internal/controller/nonadminbackup_controller.go

```diff
@@ -641,6 +641,12 @@ func (r *NonAdminBackupReconciler) createVeleroBackupAndSyncWithNonAdminBackup(c
 			backupSpec.StorageLocation = nonAdminBsl.Status.VeleroBackupStorageLocation.Name
 		}
 
+		// Exclude NAC resources (NAB, NAR, NABSL) from Non-Admin backups
+		backupSpec.ExcludedResources = append(backupSpec.ExcludedResources,
+			nacv1alpha1.NonAdminBackups,
+			nacv1alpha1.NonAdminRestores,
+			nacv1alpha1.NonAdminBackupStorageLocations)
+
```

Comment:
I just realized that this will break velero validation if the NAB spec sets a value for ExcludedNamespacedResources -- since there's velero validation to ensure that we don't include both old and new style includes/excludes. I think you need to look at the current value for backupSpec.ExcludedResources and backupSpec.ExcludedNamespaceScopedResources and append these by whichever is nonempty. Actually, you have to look at the full list of params. if all new-style params are empty, then append to ExcludedResources, else append to ExcludedNamespaceScopedResources.

Reply:
good catch

Ref https://github.com/vmware-tanzu/velero/blob/80cea31a8457835086371d9036f67ecb150208b6/pkg/controller/backup_controller.go#L460-L467

https://github.com/vmware-tanzu/velero/blob/80cea31a8457835086371d9036f67ecb150208b6/pkg/controller/backup_controller.go#L922-L932

Reply:
Maybe in the NAB we should *not* allow to use old filter parameters on the NAB validation part?

Reply:
I agree we should prioritize using new syntax, but since old syntax it is still allowed in velero, I think is ok to use it here

Reply:
From what I've seen, the vast majority of users still use the old syntax. Also, it's not deprecated, so it should be fully supported.

Reply:
It does complicate the logic here, though -- since we have to determine which syntax the user is using based on several spec fields and append to the appropriate field.

Reply:
Basically, the logic here would be
```
if haveNewResourceFilterParameters {
    append(backupSpec.ExcludedNamespaceScopedResources...
} else {
    append(backupSpec.ExcludedResources...
}
```
where `haveNewResourceFilterParameters` is defined as it is here:
https://github.com/vmware-tanzu/velero/blob/80cea31a8457835086371d9036f67ecb150208b6/pkg/controller/backup_controller.go#L926-L929

Reply:
Thanks you all ! updated the PR, PTAL !